crypto/ml_dsa/ml_dsa_key.c

```diff
@@ -312,38 +312,33 @@ int ossl_ml_dsa_key_fromdata(ML_DSA_KEY *key, const OSSL_PARAM params[],
                              int include_private)
 {
     const OSSL_PARAM *pub = NULL, *priv = NULL;
-    const uint8_t *pub_data = NULL, *priv_data = NULL;
-    size_t pub_data_len = 0, priv_data_len = 0;
 
     pub = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_PUB_KEY);
-    if (pub != NULL
-            && !OSSL_PARAM_get_octet_string_ptr(pub, (const void **)&pub_data,
-                                                &pub_data_len))
+    if (pub != NULL && pub->data_type != OSSL_PARAM_OCTET_STRING)
         return 0;
 
     /* Private key is optional */
     if (include_private) {
         priv = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_PRIV_KEY);
         if (priv != NULL) {
-            if (!OSSL_PARAM_get_octet_string_ptr(priv, (const void **)&priv_data,
-                                                 &priv_data_len)
-                    || !ossl_ml_dsa_sk_decode(key, priv_data, priv_data_len))
+            if (priv->data_type != OSSL_PARAM_OCTET_STRING
```

Comment:
sigh - thats what I had to start with...
perhaps if we had yet another function that did the calls of OSSL_PARAM_get_octet_string_ptr() in the other order then that would be sufficient?

Reply:
I did contemplate making string_ptr and string interoperate to some extent.